overlord/ifacestate/ifacestate.go

```diff
@@ -212,6 +222,8 @@ func connect(st *state.State, plugSnap, plugName, slotSnap, slotName string, fla
 	// wait for prepare-plug- anyway, and a simple one-to-one wait dependency makes testing easier.
 	addTask(connectInterface)
 	prev = connectInterface
+	// mark as the last task in connect prepare
+	tasks.MarkEdge(connectInterface, ConnectTaskEdge)
```

Comment:
should we mark the edges only if DelaySetupProfile is set?  it might be slightly clearer

Reply:
Good idea, done.